NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -150,121 +267,135 @@ NFIQ2UI::executeSingle(std::shared_ptr<BE::Image::Image> img,
 
 	cv::Mat postResample {};
 
-	if ((resolution.xRes != resolution.yRes) || (imageDPI != 500)) {
-		// Possible re-sampling
-		if (flags.force) {
-			resampled = true;
-			// Re-sample by force
-			try {
-				cv::Mat preResample { static_cast<int>(
-							  imageHeight),
-					static_cast<int>(imageWidth), CV_8U,
-					grayscaleRawData };
-				NFIR::resample(preResample, postResample,
-				    imageDPI, 500, "", "");
-
-			} catch (const cv::Exception &e) {
-				logger->printError(name, fingerPosition,
-				    "'Error: Matrix creation error: " + e.msg +
-					"'",
-				    quantized, resampled);
-				return;
+	static const uint8_t defaultDPI { 72 };
+	static const uint16_t requiredDPI { 500 };
+
+	const NFIQ2UI::DimensionInfo dimensionInfo { imageHeight, imageWidth,
+		imageDPI, requiredDPI };
 
-			} catch (const NFIR::Miscue &e) {
-				const std::string errStr { e.what() };
-				logger->printError(name, fingerPosition,
-				    "'" + errStr + "'", quantized, resampled);
+	if (resolution.xRes != resolution.yRes || imageDPI != requiredDPI) {
+		if (flags.force && imageDPI != defaultDPI) {
+			// resample by force
+			imageProps.resampled = true;
+			try {
+				postResample = NFIQ2UI::resampleAndLogError(
+				    grayscaleRawData, dimensionInfo, imageProps,
+				    logger);
+			} catch (const NFIQ2UI::ResampleError &e) {
+				if (e.errorWasHandled()) {
```

Comment:
I'd argue _you_ don't have to do this, but it doesn't hurt. Just an FYI. You know your logger isn't `nullptr` (right?).